Src/baro_LPS22HB.c

```diff
@@ -1,26 +1,80 @@
 /*
  * baro_LPS22HB.c
  *
- *  Created on: Feb 12, 2019
- *      Author: MiguelFAlvarez
  */
 
 #include "main.h"
 #include "baro_LPS22HB.h"
 #include "spi.h"
 
+static uint8_t baroInitOk=0;
+
 uint8_t baroInit()
 {
 	if(baroReadReg(REG_WHOAMI) != WHO_AM_I_LPS22HB)
+	{
+		baroInitOk=0;
 		return 0;
+	}
 
-	//baroWriteReg()
-
+	baroInitOk=1;
 	return 1;
 }
 
+uint32_t baroReadPressTemp(int16_t *temperature)
+{
+	if(!baroInitOk)
+	{
+		return 0xffffffff;
+	}
+
+	baroWriteReg(REG_CTRL_REG2,0x01);// start one shot conversion
+
+	HAL_Delay(15); // wait 15 ms
```

Comment:
I don't recommend using oneshot mode, mainly to avoid the need for delays and while loops in the code that could block other functions. It is also intended to be a low power mode where data is requested on demand rather than the sensor running continuously.
Instead, consider configuring the ODR bits in CTRL_REG1 as part of your init function. Then for your read pressure/temp function, you can just have an "if" check for the status. If status is good, do a read, otherwise skip reading and let it happen again next time the main lop tries to read. This will also mean no need for the delay within the status check while loop.

Also consider configuring EN_LPFP and LPFP_CFG settings as part of your init function